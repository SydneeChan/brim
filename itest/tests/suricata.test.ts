import path from "path"

import {handleError, stdTest} from "../lib/jest"
import appStep from "../lib/appStep/api"
import newAppInstance from "../lib/newAppInstance"

describe("Ingest tests", () => {
  let app
  let testIdx = 0
  beforeAll(() => {
    app = newAppInstance(path.basename(__filename), ++testIdx)
    return appStep.startApp(app)
  })

  afterAll(async () => {
    if (app && app.isRunning()) {
      return await app.stop()
    }
  })

  stdTest(`ingest of alerts.pcap (suricata)`, (done) => {
    appStep
      .ingestFile(app, "alerts.pcap")
      .then(async () => {
        const results = await appStep.search(
          app,
          "event_type = alert | every 1s count()"
        )
        expect(results).toMatchSnapshot()
        done()
      })
      .catch((err) => {
        handleError(app, err, done)
      })
  })
})
