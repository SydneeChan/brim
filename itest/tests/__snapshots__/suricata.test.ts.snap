// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ingest tests ingest of alerts.pcap (suricata) 1`] = `
Array [
  "ts",
  "count",
  "2015-03-05T15:07:13Z",
  "13",
  "2015-03-05T15:04:28Z",
  "2",
]
`;
